### PR TITLE
feature/543_add_missing_fields_to_ckan

### PR DESCRIPTION
**Pull request by frbattid**
* Implements issue #543 
* 100% unit tests passed:
```
Tests run: 69, Failures: 0, Errors: 0, Skipped: 0
```
* (unofficial) e2e tests passed:
```
---
row
---
time=2015-12-01T11:49:49.051CET | lvl=INFO | trans=1448966929-547-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[150] : Starting transaction (1448966929-547-0000000000)
time=2015-12-01T11:49:49.053CET | lvl=INFO | trans=1448966929-547-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[232] : Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "speed",            "type" : "float",            "value" : "112.9"          },          {            "name" : "oil-level",            "type" : "float",            "value" : "74.6"          }        ],        "type" : "car",        "isPattern" : "false",        "id" : "car1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
time=2015-12-01T11:49:49.055CET | lvl=INFO | trans=1448966929-547-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[255] : Event put in the channel (id=1018719470, ttl=10)
time=2015-12-01T11:50:02.109CET | lvl=INFO | trans=1448966929-547-0000000000 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[246] : Batch accumulation time reached, the batch will be processed as it is
time=2015-12-01T11:50:02.111CET | lvl=INFO | trans=1448966929-547-0000000000 | function=persistAggregation | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionCKANSink[383] : [ckan-sink] Persisting data at OrionCKANSink (orgName=frb, pkgName=frb_test1, resName=car1_car, data={"recvTimeTs": "1448966989","recvTime": "2015-12-01T10:49:49.55Z","fiware-servicepath": "test1","entityId": "car1","entityType": "car","attrName": "speed","attrType": "float","attrValue": "112.9"},{"recvTimeTs": "1448966989","recvTime": "2015-12-01T10:49:49.55Z","fiware-servicepath": "test1","entityId": "car1","entityType": "car","attrName": "oil-level","attrType": "float","attrValue": "74.6"})
time=2015-12-01T11:50:05.992CET | lvl=INFO | trans=1448966929-547-0000000000 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[295] : Finishing transaction (1448966929-547-0000000000)
```
ckan_test_1
```
------
column
------
time=2015-12-01T16:48:29.463CET | lvl=INFO | trans=1448984904-149-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[150] : Starting transaction (1448984904-149-0000000000)
time=2015-12-01T16:48:29.466CET | lvl=INFO | trans=1448984904-149-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[232] : Received data ({  "subscriptionId" : "51c0ac9ed714fb3b37d7d5a8",  "originator" : "localhost",  "contextResponses" : [    {      "contextElement" : {        "attributes" : [          {            "name" : "speed",            "type" : "float",            "value" : "112.9",            "metadatas": [              {                "name": "resolution",                "type": "integer",                "value": "1"              }            ]          },          {            "name" : "oil_level",            "type" : "float",            "value" : "74.6",            "metadatas": [              {                "name": "resolution",                "type": "integer",                "value": "1"              }            ]          }        ],        "type" : "car",        "isPattern" : "false",        "id" : "car1"      },      "statusCode" : {        "code" : "200",        "reasonPhrase" : "OK"      }    }  ]})
time=2015-12-01T16:48:29.469CET | lvl=INFO | trans=1448984904-149-0000000000 | function=getEvents | comp=Cygnus | msg=com.telefonica.iot.cygnus.handlers.OrionRestHandler[255] : Event put in the channel (id=2103745937, ttl=10)
time=2015-12-01T16:48:50.592CET | lvl=INFO | trans=1448984904-149-0000000000 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[246] : Batch accumulation time reached, the batch will be processed as it is
time=2015-12-01T16:48:50.593CET | lvl=INFO | trans=1448984904-149-0000000000 | function=persistAggregation | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionCKANSink[383] : [ckan-sink] Persisting data at OrionCKANSink (orgName=frb, pkgName=frb_test_column, resName=car1_car, data={"recvTime": "2015-12-01T15:48:29.469Z","fiware-servicepath": "test_column","entityId": "car1","entityType": "car","speed": "112.9","speed_md": [{"name":"resolution","type":"integer","value":"1"}],"oil_level": "74.6","oil_level_md": [{"name":"resolution","type":"integer","value":"1"}]})
time=2015-12-01T16:48:51.802CET | lvl=INFO | trans=1448984904-149-0000000000 | function=process | comp=Cygnus | msg=com.telefonica.iot.cygnus.sinks.OrionSink[295] : Finishing transaction (1448984904-149-0000000000)
```
ckan_test_2
* Assignee @fgalan

### CODE REVIEW

File: resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env python
```
Comment (fgalan) [inline resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py]: Missing license header.

File: resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+   print('where ckan_host')
+   print('      ckan_port')
+   print('      api_key')
+   print('      org_name')
+   print('      attr_persistence')
+   print('      null_value : string to be inserted as null value, use the keyword \"empty\" for an empty value')
+   print('      backup     : either true or false')
+   sys.exit(0)
+
+# Input parameters
```
Comment (fgalan) [inline resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py]: There is a nice library that I discover recently to do argument parsing. You can have a look to an example here: https://github.com/telefonicaid/fiware-orion/blob/develop/scripts/managedb/csubs_control/csub_dups_checker.py#L47

If you find it useful but you are in a rush to close this PR, I'd suggest to include a FIXME about it.

File: resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+# Do backup of a resource given its fields, its records and its id
+def do_backup(fields, records, res_name, pkg_id):
+   print('       |_Backing the resource')
+
+   # Common headers
+   headers = {'Authorization': api_key}
+
+   # Create the backup resource
+   url = 'http://' + ckan_host + ':' + ckan_port + '/api/3/action/resource_create'
```
Comment (fgalan) [inline resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py]: I think you can also the following to compose strings (similar to C):

```
url = 'http://%s:%s/api/3/action/resource_create' % (ckan_host, ckan_port)
```

Sometimes is shorter and clerarer.

File: resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py (hunk 1 above)
Comment (frbattid) [inline resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/ba085d5de84bd56ba290fcb08b9a9a7ff155e4f8

File: resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py (hunk 3 above)
Comment (frbattid) [inline resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py]: Fixed in https://github.com/telefonicaid/fiware-cygnus/commit/ba085d5de84bd56ba290fcb08b9a9a7ff155e4f8

File: resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py (hunk 2 above)
Comment (frbattid) [inline resources/cygnus-translators/cygnus_translator_pre0.10.0_to_0.10.0_ckan.py]: FIXME added in https://github.com/telefonicaid/fiware-cygnus/commit/eec53a751e3aa069cff67d7b7e830892a32f1fc1

**Comment by fgalan** [comment]
LGTM

**Pull request merged**